datalad_dataverse/remote.py

```diff
@@ -77,18 +80,34 @@ def checkpresent(self, key):
         dataset = self.api.get_dataset(identifier=self.doi)
 
         datafiles = dataset.json()['data']['latestVersion']['files']
-        if next((item for item in datafiles if item['label'] == key), None):
+        if next((item for item in datafiles if item['dataFile']['filename'] == key), None):
             return True
         else:
             return False
 
-    def transfer_store(self, key, local_file):
+    def checkpresentexport(self, key, remote_file):
+        return self.checkpresent(key=remote_file)
+
+    def transfer_store(self, key, local_file, datafile=None):
         ds_pid = self.doi
+        if datafile is None:
+            datafile = Datafile()
+            datafile.set({'filename': key, 'label': key})
+        datafile.set({'pid': ds_pid})
+        
+        resp = self.api.upload_datafile(identifier=ds_pid, filename=local_file, json_str=datafile.json())
+        resp.raise_for_status()
+
+    def transferexport_store(self, key, local_file, remote_file):
+        if re.search(pattern='[^a-z0-9_\-.\\/\ ]', string=os.path.dirname(remote_file), flags=re.ASCII | re.IGNORECASE):
```

Comment:
Call to `os.path.dirname()` is duplicated below, better to do it once, and use a variable.

Reply:
Fixed now :)